folium/features.py

```diff
@@ -725,93 +885,88 @@ class GeoJsonTooltip(Tooltip):
-            keys = tuple(self._parent.data['objects'][obj_name][
-                             'geometries'][0]['properties'].keys())
-        else:
-            raise TypeError('You cannot add a GeoJsonTooltip to anything else '
-                            'than a GeoJson or TopoJson object.')
-        keys = tuple(x for x in keys if x not in ('style', 'highlight'))
-        for value in self.fields:
-            assert value in keys, ('The field {} is not available in the data. '
-                                   'Choose from: {}.'.format(value, keys))
-        super(GeoJsonTooltip, self).render(**kwargs)
+    Also has functionality to embed dynamic VegaLite charts for each feature based on a 'join' key for an attribute of
+    the GeoJson feature, and a data layer in a VegaLite specification. Make sure both join keys contain string
+    datatypes for best consistency.
+
+    Parameters
+    ----------
+    fields: list or tuple.
+        Labels of GeoJson/TopoJson 'properties' or GeoPandas GeoDataFrame
+        columns you'd like to display.
+    aliases: list/tuple of strings, same length/order as fields, default None.
+        Optional aliases you'd like to display in the tooltip as field name
+        instead of the keys of `fields`.
+    labels: bool, default True.
+        Set to False to disable displaying the field names or aliases.
+    localize: bool, default False.
+        This will use JavaScript's .toLocaleString() to format 'clean' values
+        as strings for the user's location; i.e. 1,000,000.00 comma separators,
+        float truncation, etc.
+        *Available for most of JavaScript's primitive types (any data you'll
+        serve into the template).
+    style: str, default None.
+        HTML inline style properties like font and colors. Will be applied to
+        a div with the text in it.
+    vegalite: folium.VegaLite, default None.
+        folium wrapper for a VegaLite chart specification.
+    map_key: str, default None.
+        A key in the GeoJson properties to base the join on.
+    data_key: str, default None.
+        A key value in the VegaLite data layer to base the join on.
+
+    Examples
+    ---
+    gjson = folium.GeoJson(gdf).add_to(m)
+
+    folium.features.GeoJsonPopup(fields=['NAME'],
+                                labels=False,
+                                vegalite=chart,
+                                map_key='county_code',
+                                data_key='fips'
+                                ).add_to(gjson)
+    """
+
+    _template = Template(u"""
+        {% macro script(this, kwargs) %}
+        {{ this._parent.get_name() }}.bindPopup(""" +
+                         GeoJsonDetail.base_template +
+                         u""",{{ this.popup_options | tojson | safe }})
+                         {% endmacro %}""")
+
+    def __init__(self, fields=None, aliases=None, labels=True, style="margin: auto;", vegalite=None, map_key=None,
```

Comment:
What do you think about giving this popup the same options as the 'regular' Popup class? I mean `show` and `sticky` options.